compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1882,42 +1881,29 @@ private void checkDuplicateVarRefs(List<BLangExpression> varRefs) {
 
     private void checkDuplicateVarRefs(List<BLangExpression> varRefs, Set<BSymbol> symbols) {
         for (BLangExpression varRef : varRefs) {
-            if (varRef == null) {
-                continue;
-            }
             NodeKind kind = varRef.getKind();
-            if (kind != NodeKind.SIMPLE_VARIABLE_REF
-                    && kind != NodeKind.RECORD_VARIABLE_REF
-                    && kind != NodeKind.ERROR_VARIABLE_REF
-                    && kind != NodeKind.TUPLE_VARIABLE_REF) {
-                continue;
-            }
-
-            if (kind == NodeKind.SIMPLE_VARIABLE_REF
-                    && names.fromIdNode(((BLangSimpleVarRef) varRef).variableName) == Names.IGNORE) {
-                continue;
-            }
-
-            if (kind == NodeKind.TUPLE_VARIABLE_REF) {
-                checkDuplicateVarRefs(getVarRefs((BLangTupleVarRef) varRef), symbols);
+            if (kind == NodeKind.SIMPLE_VARIABLE_REF) {
```

Comment:
A switch will be more efficient since if it's TUPLE_VARIABLE_REF, it have to check 4 if-checks

Reply:
Replace this with a switch case